web_src/js/components/RepoActionView.vue

```diff
@@ -425,6 +431,23 @@ const sfc = {
       } else {
         actionBodyEl.append(fullScreenEl);
       }
+    },
+    async hashChangeListener() {
+      const selectedLogStep = window.location.hash;
+      const [_, step, _line] = selectedLogStep.split('-');
+      if (!this.currentJobStepsStates[step]) return;
+      if (!this.currentJobStepsStates[step].expanded && this.currentJobStepsStates[step].cursor === null) {
+        this.currentJobStepsStates[step].expanded = true;
+        // need to await for load job if the step log is loaded for the first time
+        // so logline can be selected by querySelector
+        await this.loadJob();
+      }
+      const logline = this.$refs.steps.querySelector(`${selectedLogStep}`);
+      if (!logline) return;
+      const logSummary = this.$refs.steps.querySelector(`.job-step-section:nth-of-type(${parseInt(step) + 1}) > .job-step-summary`);
+      const offset = logSummary.offsetHeight + document.querySelector('.job-info-header').offsetHeight;
+      logline.querySelector('.line-num').click();
+      window.scrollTo({top: logline.offsetTop - offset, behavior: 'instant'});
```

Comment:
Maybe use `logline.scrollIntoView({behavior: 'instant'})`, it should not need this offset calculation.

Reply:
> * `element.scrollIntoView` is not working as expected (did not take fixed elements into account), so calculated the heights of fixed elements and used `window.scrollTo`.
> 

I tried this at the first place, but it will scroll under the fixed elements, e.g.,

<img width="1427" alt="Screen Shot 2023-06-28 at 08 44 51" src="https://github.com/go-gitea/gitea/assets/17645053/c51abfc1-634c-4a4e-84ed-a2af176973a8">

Reply:
Does the `block` option help? Where should it scroll to?

> Defines vertical alignment. One of start, center, end, or nearest. Defaults to start.

Reply:
It works, updated to use `logLine.scrollIntoView({behavior: 'instant', block: 'center'});` and updated demo video